src/main/webapp/app/exercises/shared/feedback/feedback.service.ts

```diff
@@ -0,0 +1,188 @@
+import { Injectable } from '@angular/core';
+import { Feedback, FeedbackType, STATIC_CODE_ANALYSIS_FEEDBACK_IDENTIFIER, SUBMISSION_POLICY_FEEDBACK_IDENTIFIER } from 'app/entities/feedback.model';
+import { FeedbackItem, FeedbackItemType } from 'app/exercises/shared/result/result-detail.component';
+import { TranslateService } from '@ngx-translate/core';
+import { computeFeedbackPreviewText } from 'app/exercises/shared/feedback/feedback.util';
+import { StaticCodeAnalysisIssue } from 'app/entities/static-code-analysis-issue.model';
+
+@Injectable({ providedIn: 'root' })
+export class FeedbackService {
+    constructor(private translateService: TranslateService) {}
+
+    /**
+     * Creates a feedback item with a category, title and text for each feedback object.
+     * @param feedbacks The list of feedback objects.
+     * @param isProgrammingExercise
+     * @param showTestDetails
+     */
+    public createFeedbackItems(feedbacks: Feedback[], isProgrammingExercise: boolean, showTestDetails: boolean): FeedbackItem[] {
```

Comment:
As this was moved out of the component, we need to pass `isProgrammingExercise: boolean, showTestDetails: boolean` as method parameters. I personally dislike this solution. Any suggestions here?